gustaf/spline/create.py

```diff
@@ -0,0 +1,229 @@
+"""gustaf/spline/create.py
+
+Create operations for spline geometries
+"""
+
+
+import itertools
+import logging
+import numpy as np
+
+
+def extrude(spline, axis=None):
+    """
+    Extrudes Splines ype objects of any spline-type object
+
+    Parameters
+    ----------
+    spline : Gustaf-Spline
+    axis : Extrusion axis
+    """
+    from gustaf.spline.base import GustafSpline
+
+    # Check input type
+    from gustaf.spline.base import GustafSpline
+    if not issubclass(type(spline), GustafSpline):
+        raise NotImplementedError("Extrude only works for splines")
+
+    # Check axis
+    if axis is not None:
+        axis = np.asarray(axis)
+        if len(axis.shape) == 0:
+            axis = np.asarray([axis])
+    else:
+        raise ValueError("No extrusion Axis given")
+
+    # Check Axis dimension
+    if not (spline.control_points.shape[1] == axis.shape[0]):
+        raise ValueError(
+            "Dimension Mismatch between extrusion axis and spline."
+        )
```

Comment:
```suggestion
    # Check Axis dimension
    # formulate correct cps
    if spline.dim == axis.shape[0]:
        cps = spline.control_points
    elif int(spline.dim + 1) == axis.shape[0]:
        # one smaller dim is allowed
        # warn that we assume new dim is all zero
        utils.log.warning(
            "Given axis is one dimension bigger than spline's dim.",
            "Assuming 0.0 entries for new dimension."
        )
        cps = np.hstack(
            (
                spline.control_points,
                np.ones((len(spline.control_points), 1)
            )
        )
    else:
        raise ValueError(
            "Dimension Mismatch between extrusion axis and spline."
        )
```

Reply:
This is pretty clever, but should be zero. Also ... Why not more?